packages/aws-cdk-lib/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -150,7 +151,7 @@ class StepFunctionsExecutionIntegration extends AwsIntegration {
     if (this.stateMachine instanceof sfn.StateMachine) {
       const stateMachineType = (this.stateMachine as sfn.StateMachine).stateMachineType;
       if (stateMachineType !== sfn.StateMachineType.EXPRESS) {
-        throw new Error('State Machine must be of type "EXPRESS". Please use StateMachineType.EXPRESS as the stateMachineType');
+        throw new ValidationError('State Machine must be of type "EXPRESS". Please use StateMachineType.EXPRESS as the stateMachineType', this.stateMachine);
```

Comment:
This is more appropriate I think. Yes the error is _about_ the State Machine. But `bind`ing it to the `Method` is what's causing the error. Or in other words, this is about location of the error in the tree.

```suggestion
        throw new ValidationError('State Machine must be of type "EXPRESS". Please use StateMachineType.EXPRESS as the stateMachineType', method);
```